react-game/src/redux/game-reducer.ts

```diff
@@ -0,0 +1,416 @@
+                return e.content === cardPair[0].content || e.content === cardPair[1].content
+                    ? {...e, type: type}
+                    : e
+            }
+            return {
+                ...state,
+                board: state.board.map((e: CardType) =>
+                    e.type === "open"
+                        ? condition(e, type)
+                        : e.type === "wrong" ? condition(e, type) : e
+                )
+            }
+        }
+        case "SET_GAME_TYPE": {
+            localStorage.setItem("gameType", JSON.stringify(action.gameType))
+            return {...state, gameType: action.gameType}
+        }
+        case "SET_BOARD_SIZE": {
+            localStorage.setItem("boardSize", `${action.boardSize}`)
+            return {...state, boardSize: action.boardSize}
+        }
+        case "DISABLE_BOARD": {
+            return {...state, boardDisabled: action.boolean}
+        }
+        case "SET_SOUNDS_VOLUME": {
+            return {...state, soundsVolume: action.volume}
+        }
+        case "SET_MUSIC_VOLUME": {
+            return {...state, musicVolume: action.volume}
+        }
+        case "SET_CARD_PAIR": {
+            return {...state, cardPair: [...state.cardPair, action.card]}
+        }
+        case "SET_CARD_STYLE": {
+            return {...state, imgURL: action.imgURL}
+        }
+        case "SET_CARD_PAIRS": {
+            return {...state, cardPair: [...action.cards]}
+        }
+        case "SET_DIFFICULTY": {
+            localStorage.setItem("difficulty", `${action.difficulty}`)
+            return {...state, difficulty: action.difficulty}
+        }
+        case "SET_TIMER": {
+            return {...state, time: action.time}
+        }
+        case "SET_SCORE": {
+            return {...state, score: action.score}
+        }
+        case "SET_HIGH_SCORE": {
+            return {...state, highScore: action.score}
+        }
+        case "SET_GAMES": {
+            return {...state, games: [action.game, ...state.games]}
+        }
+        case "SET_ALL_GAMES": {
+            return {...state, games: action.games}
+        }
+        case "SET_AUTO_PLAY": {
+            return {...state, autoPlay: action.autoPlay}
+        }
+        case "CLEAR_CARD_PAIR": {
+            return {...state, cardPair: []}
+        }
+        default:
+            return state
+    }
+};
+
+type ActionsType =
+    openCardType
+    | setBoardType
+    | setAnswersType
+    | disableBoardType
+    | clearCardPairType
+    | setCardPairType
+    | setGameType
+    | setCardPairsType
+    | setSoundsVolumeType
+    | setMusicVolumeType
+    | setBoardSizeType
+    | setCardStyleType
+    | setDifficultyType
+    | setScoreType
+    | setTimerType
+    | setHighScoreType
+    | setGamesType
+    | setAllGamesType
+    | setAutoPlayType
+
+
+export const openCard = (number: number): openCardType => ({
```

Comment:
![image](https://user-images.githubusercontent.com/43242417/110517229-43653500-8113-11eb-8779-d6d5ff977e76.png)


There's a good pattern to split reducers, actions, selectors, middlewares. And to have the structure like that.